src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metrics_experience_grid.tsx

```diff
@@ -100,17 +99,28 @@ export const MetricsExperienceGrid = ({
   }, [valueFilters]);
 
   if (fields.length === 0) {
-    return <EmptyState isLoading={isLoading} />;
+    return (
+      <MetricsGridHeader
+        indexPattern={indexPattern}
+        renderToggleActions={renderToggleActions}
+        chartToolbarCss={chartToolbarCss}
+        requestParams={requestParams}
+        setDebouncedSearchTerm={setDebouncedSearchTerm}
+        fields={fields}
+      >
+        <EmptyState isLoading={isLoading} />
+      </MetricsGridHeader>
```

Comment:
q: why do we need to wrap the empty state with `<MetricsGridHeader />`?

Reply:
So, the toolbar problem that we had should've been fixed by this PR https://github.com/elastic/kibana/pull/234616. The empty state will only cover the entire grid section if the Fields API returns empty. If the dimensions selected or the search causes don't match any metrics in the grid, the toolbar should still be visible without this wrapper.

Reply:
That's true, as you mentioned, now with those changes, it works without the wrapper. I removed it from the empty state. 👍